packages/node/src/integrations/tracing/nest/nest.ts

```diff
@@ -87,10 +87,11 @@ export function setupNestErrorHandler(app: MinimalNestJsApp, baseFilter: NestJsE
         const originalCatch = Reflect.get(target, prop, receiver);
 
         return (exception: unknown, host: unknown) => {
-          const status_code = (exception as { status?: number }).status;
+          const status_code = (exception as ExpectedException).status;
+          const error_property = (exception as ExpectedException).error;
 
           // don't report expected errors
-          if (status_code !== undefined) {
+          if (status_code !== undefined || error_property !== undefined) {
             return originalCatch.apply(target, [exception, host]);
           }
 
```

Comment:
h: Let's go for something like the following here:

```ts
const exceptionIsObject = typeof exception === 'object' && exception !== null;
const exceptionStatusCode = exceptionIsObject && 'status' in exception ? exception.status : null;
const exceptionErrorProperty = exceptionIsObject && 'error' in exception ? exception.error : null;

// Only report expected errors
// - ExpectedNestJS control flow errors that with status codes like NestJS' `HttpException` errors will have a `status` property.
// - NestJS' expected `RpcExceptions` will have an `error` property
if (exceptionStatusCode === null && exceptionErrorProperty === null) {
    captureException(exception, {
        mechanism: {
        handled: false,
        },
    });
}

return originalCatch.apply(target, [exception, host]);
```

Reply:
Why? It's not immediately obvious to me how that improves the current implementation

Reply:
The current implementation will throw if the error is nullish, which can always happen. We should avoid doing any type casts (` as `) as much as possible. The change I suggested does runtime checks not to throw when you 

Also, we should add documentation. Somebody who comes across the code is basically in the dark what it means to look on the `status` or `error` props.

Reply:
Ok sure, thanks for clarifying. Will update